compiler/src/lexer.mll

```diff
@@ -134,10 +134,10 @@ rule main = parse
   | '"' (([^'"' '\\']|'\\' _)* as s) '"' { STRING (unescape (L.of_lexbuf lexbuf) s) }
 
   (* Why this is needed *)
-  | ((*'-'?*) digit+) as s   
-      {INT s} 
+  | ((*'-'?*) digit+('_'digit+)*) as s   
+      {INT Str.(global_replace (regexp "_") "" s)} 
 
-  | ('0' ['x' 'X'] hexdigit+) as s
+  | ('0' ['x' 'X' 'b' 'B' 'o' 'O'] hexdigit+) as s
```

Comment:
Are we sure we want to accept `0babcd` and `0o99` as valid literals?

Reply:
I corrected it in the last commit, I didn't saw the error at the time